src/cabinDatabase/cabinDatabase.api.ts

```diff
@@ -0,0 +1,49 @@
+import { Injectable, OnModuleInit } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import {
+  createClient,
+  PostgrestResponse,
+  SupabaseClient,
+} from '@supabase/supabase-js';
+import { SupabaseCabin } from './cabinDatabase.interface';
+
+@Injectable()
+export class CabinDatabaseApi implements OnModuleInit {
+  private supabase: SupabaseClient;
+  private CABIN_REQUEST_DEFAULT_LIMIT = 1000; // TODO 1000 is an artificial limit to prevent limit undefined
+
+  constructor(private readonly configService: ConfigService) {}
+
+  onModuleInit() {
+    const supabaseUrl = this.configService.getOrThrow('SUPABASE_URL');
+    const supabaseKey = this.configService.getOrThrow('SUPABASE_KEY');
+    this.supabase = createClient(supabaseUrl, supabaseKey);
+  }
+
+  async upsertCabins(cabins: Partial<SupabaseCabin>[]) {
+    const errors = [];
+    for (const cabin of cabins) {
+      const { error } = await this.supabase.from('cabins').upsert(cabin, {});
+      if (error !== null) {
+        errors.push(error);
+      }
+    }
+
+    return errors;
+  }
+
+  async getRandomBookableCabins(
+    limit = this.CABIN_REQUEST_DEFAULT_LIMIT,
+  ): Promise<PostgrestResponse<SupabaseCabin>> {
+    return await this.supabase
+      .rpc('get_random_cabins')
```

Comment:
Where is this RPC call defined? In Supabase?

Reply:
Yep, it's a database function that I registered manually.

I found supabase js quite limiting, so we may want to change to a direct connection so we can build queries more dynamically. If I understood correctly, the database function would then be obsolete.

Reply:
I suggest that for a future PR and will create an issue from it.